Dockerfile

```diff
@@ -1,13 +1,35 @@
 # Icinga DB Docker image | (c) 2020 Icinga GmbH | GPLv2+
 
-FROM golang:alpine as entrypoint
-RUN ["sh", "-exo", "pipefail", "-c", "apk add upx; rm -vf /var/cache/apk/*"]
-COPY entrypoint /entrypoint
+FROM golang:alpine as go-upx
+RUN ["sh", "-exo", "pipefail", "-c", "apk add git upx; rm -vf /var/cache/apk/*"]
+ENV CGO_ENABLED 0
+
+
+FROM go-upx as icingadb
+
+COPY --from=icingadb-git . /icingadb-src/.git
+WORKDIR /icingadb-src
+RUN ["git", "checkout", "."]
+
+RUN --mount=type=cache,target=/go/pkg/mod \
+    --mount=type=cache,target=/root/.cache/go-build \
+    ["go", "build", "-ldflags", "-s -w", "./cmd/icingadb"]
+
+RUN ["upx", "icingadb"]
 
+RUN ["bzip2", "-k", "schema/mysql/schema.sql"]
+RUN ["bzip2", "-k", "schema/pgsql/schema.sql"]
+
+
+FROM go-upx as entrypoint
```

Comment:
Is there a benefit in using multiple build stages? It's the same base container, so it should be just fine to build Icinga DB and the entry point in the same container.

Reply:
Yes, caching. Only changed stuff gets re-built.